packages/story-editor/src/components/autoSaveHandler/index.js

```diff
@@ -26,30 +27,34 @@ import { useConfig, useHistory, useStory } from '../../app';
 import useIsUploadingToStory from '../../utils/useIsUploadingToStory';
 
 function AutoSaveHandler() {
+  const improvedAutosaves = useFeature('improvedAutosaves');
   const { autoSaveInterval } = useConfig();
   const {
     state: { hasNewChanges },
   } = useHistory();
-  const { status, saveStory } = useStory(
+  const { status, saveStory, autoSave } = useStory(
     ({
       state: {
         story: { status },
       },
-      actions: { saveStory },
+      actions: { autoSave, saveStory },
     }) => ({
       status,
+      autoSave,
       saveStory,
     })
   );
   const isUploading = useIsUploadingToStory();
 
   const isDraft = 'draft' === status || !status;
 
+  const save = improvedAutosaves ? autoSave : saveStory;
```

Comment:
Hm, would it still be `saveStory` for a draft always? Or is this handled in the backend?

Reply:
Looks like there's a isDraft check here

https://github.com/GoogleForCreators/web-stories-wp/blob/339cb6598fc1b3d3b0b99b33c373c159c8f8f1bd/packages/story-editor/src/components/autoSaveHandler/index.js#L60

I haven't dug into the full auto save flow yet

Reply:
As far as I understand, for drafts, the story itself gets overwritten by the autosave (`saveStory`), however, in case of published stories, an autosave (`revision`) gets created instead and the story itself will not get overwritten. So I'm wondering if it should be `autoSave` only if `improvedAutoSaves && !isDraft`( That's actually something that could also be done in #1402 )?

Shouldn't this PR also include the backend logic for making sure `autoSave` is creating a revision with all the necessary data? Or is the backend logic already in place for that when `autoSave` is used here?

Reply:
I don't think we need any backend changes here. We already have a `Stories_Autosaves_Controller` and WordPress chooses between autosaves and revisions automatically depending on if it's a draft or not:

https://github.com/WordPress/wordpress-develop/blob/37368555a5251e49b748bf91407a24fad6d32e9a/src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php#L223-L230

But should be easy to verify :-)

Reply:
> In AutoSaveHandler, use autoSave instead of saveStory if flag is enabled

Seeing the autosave call 

<img width="1267" alt="Screen Shot 2022-08-17 at 5 18 06 PM" src="https://user-images.githubusercontent.com/62242/185246187-d2a9b770-11d6-4d21-9f4e-eb0957ade0c8.png">

Reply:
Alright, I wasn't sure that autosaving already saves all the data we need for stories, but that's great if it does :)